src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -193,6 +223,19 @@ public User updateUser(User user) {
     return userToBeUpdated;
   }
 
+  public void updatePreferences(User user){
+      //Getting the correct user (404 and 409 should be check by specific access already)
+      User userToUpdatePreferences = getUserById(user.getId());
+      //Update User Preferences
+      if(user.getMinAge()>= 18 && user.getMinAge() <= user.getMaxAge()){ //they can both be 22 for instance. If you only want people that are 22 years old
+          userToUpdatePreferences.setMinAge(user.getMinAge());
+          userToUpdatePreferences.setMaxAge(user.getMaxAge());
+      }
+      if(!Objects.isNull(user.getGenderPreferences()) || user.getGenderPreferences().isEmpty()){
```

Comment:
if user.getGenderPreferences() is null, the first condition evaluates to false and the second is checked, which would result in an error, since null does not have a method isEmpty(). I think what you want is 
`!Objects.isNull(user.getGenderPreferences()) && !user.getGenderPreferences().isEmpty()`
or equivalently
`! (Objects.isNull(user.getGenderPreferences()) || user.getGenderPreferences().isEmpty())`

Reply:
Updated using :!Objects.isNull(user.getGenderPreferences()) && !user.getGenderPreferences().isEmpty()